optaplanner-core/src/main/java/org/optaplanner/core/config/util/RhinoSupport.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.config.util;
+
+import java.util.function.IntSupplier;
+
+import org.mozilla.javascript.Context;
+import org.mozilla.javascript.ContextFactory;
+import org.mozilla.javascript.RhinoException;
+import org.mozilla.javascript.Scriptable;
+
+/**
+ * This class exists separately, as Rhino is an optional dependency and the imports would have immediately triggered
+ * {@link ClassNotFoundException}s if present on {@link ConfigUtils} directly.
+ */
+final class RhinoSupport implements IntSupplier {
```

Comment:
Naming: "Support" is too generic.
How about "RhinoScriptEngineWorkaround" (RhinoScriptEngine is wrong, it doesn't extend ScriptEngine) or "RhinoJavaScriptPoolSizer"?